arbnode/transaction_streamer.go

```diff
@@ -684,9 +760,44 @@ func messageFromTxes(header *arbos.L1IncomingMessageHeader, txes types.Transacti
 	}, nil
 }
 
+// The caller must hold the insertionMutex
+func (s *TransactionStreamer) resequenceReorgedMessages(messages []*arbstate.MessageWithMetadata) {
+	for _, msg := range messages {
+		// Check if the message is non-nil just to be safe
+		if msg == nil || msg.Message == nil || msg.Message.Header == nil {
+			continue
+		}
+		header := msg.Message.Header
+		if header.RequestId != nil {
+			// This is a delayed message
+			continue
```

Comment:
that was my first thought as well.. but shouldn't we try to re-sequence delayed message with the same index as well, if it was not already sequenced? Otherwise it would effectively move all delayed messages not forced by timestamp to be scheduled after all non-delayed.

Reply:
I've changed it to try to check if the delayed message was unaffected by the reorg, and if so, resequence it